client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_SET_FOCUS,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_SET_NEXT_FOCUS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+import layoutFocusModule from 'lib/layout-focus';
+
+const initialState = { current: null, previous: null, next: null };
+
+export default function layoutFocus( state = initialState, action ) {
+	const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+	const isValidArea = area => includes( validAreas, area );
+	switch ( action.type ) {
+		case LAYOUT_SET_FOCUS:
+			if ( ! isValidArea( action.area ) ) {
+				return state;
+			}
+			// Temporary while replacing module with this reducer
+			layoutFocusModule.set( action.area );
```

Comment:
I think in the case of `layoutFocus` it's not as big an issue since it's present everywhere anyways, but generally my worry with calling on modules from reducer (aside from side-effects no-no) is that reducers are loaded by boot and so are present in every section of the app. If this were a module more specific to a particular section, that would unnecessarily bloat the `commons` bundle.